src/main/java/com/entropyteam/entropay/employees/services/EmployeeService.java

```diff
@@ -69,17 +69,19 @@ protected BaseRepository<Employee, UUID> getRepository() {
     protected EmployeeDto toDTO(Employee entity) {
         List<PaymentInformation> paymentInformationList =
                 paymentRepository.findAllByEmployeeIdAndDeletedIsFalse(entity.getId());
-        List<Assignment> assignments =
-                assignmentRepository.findAssignmentByEmployee_IdAndDeletedIsFalse(entity.getId());
+        Optional<Assignment> assignments =
+                assignmentRepository.findAssignmentByEmployeeIdAndActiveIsTrueAndDeletedIsFalse(entity.getId());
         Optional<Assignment> lastAssignment = assignments.stream().filter(a -> a.getEndDate() == null).findFirst();
```

Comment:
findAssignmentByEmployeeIdAndActiveIsTrueAndDeletedIsFalse will return the active assignment, so this is the last assignment.
You could avoid line 74

Reply:
Ready, I took it out